### ISSUE TITLE
Define `PSA_WANT_xxx_KEY_PAIR_yyy` step 2/ECC

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#7771]
Following #7614 which introduced new `PSA_WANT_xxx_KEY_PAIR_yyy`, we want to remove uses of the internal `MBEDTLS_PSA_zzz_KEY_TYPE_ECC_KEY_PAIR_LEGACY` (where `zzz` is one or `WANT`, `BUILTIN`) symbol it introduced for ECC.

- [ ] Locate the code related to ECC keypair derivation in `psa_crypto.c` and/or `psa_crypto_ecp.c` and replace the `LEGACY` guard with the appropriate `_DERIVE` macro.
- [ ] Locate tests related to ECC keypair derivation and replace the `LEGACY` guard with the appropriate `_DERIVE` macro.
- [ ] Locate the code related to ECC keypair (random) generation in `psa_crypto.c` and/or `psa_crypto_ecp.c` and  replace the `LEGACY` guard with the appropriate `_GENERATE` macro.
- [ ] Locate tests related to ECC keypair derivation and replace the `LEGACY` guard with the appropriate `_GENERATE` macro.
- [ ] Create new internal macros `MBEDTLS_PSA_zzz_KEY_TYPE_ECC_KEY_PAIR_BASIC` (where `zzz` is one or `WANT`, `BUILTIN`), auto-enabled as soon as the corresponding `USE` macro is (which itself is auto-enabled as soon as anything related to the key type is), and replace all remaining uses of `LEGACY` macros with these macros. (See note below.)
- [ ] Make sure there are components in `all.sh` using `MBEDTLS_PSA_CRYPTO_CONFIG`, one enabling only `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_GENERATE` another only `_DERIVE`, a 3rd one enabling only `USE`, and that they pass `make test`.

Note: the internal `_BASIC` macro reflect that fact that in the short term we're not making it a goal to enable builds without support for import-export, as this is heavily used in tests. We are allowed to include more in the build than what's explicitly requested. In the short term, we want to include support for generate and derive only when explicitly requested, but we'll include support for import-export any time anything else is requested (or equivalently, any time `USE` is requested).

Edit: in the meantime we've decided to rename `PSA_WANT_xxx_KEY_PAIR_USE` to `PSA_WANT_xxx_KEY_PAIR_BASIC`, so I think `MBEDTLS_PSA_WANT_xxx_KEY_PAIR_BASIC` is no longer a great name for the internal macro, BASIC wouldn't mean the same thing for both. At the risk of ending up with long name, I suggest being fully explicit and calling the internal macro `MBEDTLS_PSA_WANT_xxx_KEY_PAIR_BASIC_IMPORT_EXPORT`: the user requested basic, and we translated that to basic+import+export internally since supporting only basic without import-export is not a goal for use right now.

**Depends on:** #7614
